workers/docker-worker/src/upload_to_s3.js

```diff
@@ -50,7 +50,7 @@ module.exports = async function uploadToS3 (
     // write the source out to a temporary file so that it can be
     // re-read into the request repeatedly
     if (typeof source === 'string') {
-      await tmp.writeFile(source);
```

Comment:
hmm.. this is generally a bad idea to use sync functions as it blocks whole eventloop. But I see that this `temporary` library does not handle async writes properly.

I guess for now we can go with this solution, but we should aim to rewrite this using native `fs` functions

Reply:
Just made https://github.com/taskcluster/taskcluster/issues/5287. Thanks for the notes @lotas!

Reply:
Thank you!